src/MiniProfiler.Providers.PostgreSql/PostgreSqlStorage.cs

```diff
@@ -0,0 +1,489 @@
+﻿using Dapper;
+using Npgsql;
+using StackExchange.Profiling.Internal;
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace StackExchange.Profiling.Storage
+{
+    /// <summary>
+    /// Understands how to store a <see cref="MiniProfiler"/> to a PostgreSQL Server database.
+    /// </summary>
+    public class PostgreSqlStorage : DatabaseStorageBase
+    {
+        /// <summary>
+        /// The schema that the tables are stored in. 
+        /// </summary>
+        public readonly string Schema = null;
+
+
+        private readonly string MiniProfilersTableWithSchema;
+        private readonly string MiniProfilerTimingsTableWithSchema;
+        private readonly string MiniProfilerClientTimingsTableWithSchema;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PostgreSqlStorage"/> class with the specified connection string.
+        /// </summary>
+        /// <param name="connectionString">The connection string to use.</param>
+        /// <param name="schema">The schema that the tables are stored in.</param>
+        public PostgreSqlStorage(string connectionString, string schema = null) : base(connectionString)
+        {
+            Schema = schema;
+            if (Schema != null)
+            {
+                MiniProfilersTableWithSchema = $"{Schema}.{MiniProfilersTable}";
+                MiniProfilerTimingsTableWithSchema = $"{Schema}.{MiniProfilerTimingsTable}";
+                MiniProfilerClientTimingsTableWithSchema = $"{Schema}.{MiniProfilerClientTimingsTable}";
+            }
+            else
+            {
+                MiniProfilersTableWithSchema = MiniProfilersTable;
+                MiniProfilerTimingsTableWithSchema = MiniProfilerTimingsTable;
+                MiniProfilerClientTimingsTableWithSchema = MiniProfilerClientTimingsTable;
+            }
```

Comment:
This will be a moot point in context of removing this constructor, but in the `schema == null` case, `*WithSchema` is a misnomer - it's still a table without schema, relative to the logged in SQL user, etc.